lib/ret/storage.ex

```diff
@@ -337,6 +337,111 @@ defmodule Ret.Storage do
+           file_id = Regex.run(~r/files\/([^.]*)\./, src)
+             |> Enum.at(1)
+           # Question: Use OwnedFile.set_state() instead?
+           # Question: Is it possible that an OwnedFile is referred by multiple RoomObjects?
+           from(f in OwnedFile, where: f.owned_file_uuid == ^file_id)
+             |> Repo.update_all(set: [state: :active])
+         end)
+
+    # 3. Mark the OwnedFiles as active that referred by
+    #    AppConfig, Asset, Avatar, AvatarListing, Project, Scene, or SceneListing.
+
+    from(a in Ret.AppConfig, select: a.owned_file_id)
+      |> Repo.all()
+      |> Enum.each(fn (file_id) ->
+           from(f in OwnedFile, where: f.owned_file_uuid == ^file_id)
+             |> Repo.update_all(set: [state: :active])
+         end)
+
+    from(a in Ret.Asset, select: [a.asset_owned_file_id, a.thumbnail_owned_file_id])
+      |> Repo.all()
+      |> Enum.each(fn (file_ids) ->
+           from(f in OwnedFile, where: f.owned_file_uuid in ^file_ids)
+             |> Repo.update_all(set: [state: :active])
+         end)
+
+    from(a in Ret.Avatar, select: [a.gltf_owned_file_id, a.bin_owned_file_id, a.thumbnail_owned_file_id, a.base_map_owned_file_id, a.emissive_map_owned_file_id, a.normal_map_owned_file_id, a.orm_map_owned_file_id])
+      |> Repo.all()
+      |> Enum.each(fn (file_ids) ->
+           from(f in OwnedFile, where: f.owned_file_uuid in ^file_ids)
+             |> Repo.update_all(set: [state: :active])
+         end)
+
+    from(a in Ret.AvatarListing, select: [a.gltf_owned_file_id, a.bin_owned_file_id, a.thumbnail_owned_file_id, a.base_map_owned_file_id, a.emissive_map_owned_file_id, a.normal_map_owned_file_id, a.orm_map_owned_file_id])
+      |> Repo.all()
+      |> Enum.each(fn (file_ids) ->
+           from(f in OwnedFile, where: f.owned_file_uuid in ^file_ids)
+             |> Repo.update_all(set: [state: :active])
+         end)
+
+    from(p in Ret.Project, select: [p.project_owned_file_id, p.thumbnail_owned_file_id])
+      |> Repo.all()
+      |> Enum.each(fn (file_ids) ->
+           from(f in OwnedFile, where: f.owned_file_uuid in ^file_ids)
+             |> Repo.update_all(set: [state: :active])
+         end)
+
+    from(s in Ret.Scene, select: [s.model_owned_file_id, s.screenshot_owned_file_id, s.scene_owned_file_id])
+      |> Repo.all()
+      |> Enum.each(fn (file_ids) ->
+           from(f in OwnedFile, where: f.owned_file_uuid in ^file_ids)
+             |> Repo.update_all(set: [state: :active])
+         end)
+
+    from(s in Ret.SceneListing, select: [s.model_owned_file_id, s.screenshot_owned_file_id, s.scene_owned_file_id])
+      |> Repo.all()
+      |> Enum.each(fn (file_ids) ->
+           from(f in OwnedFile, where: f.owned_file_uuid in ^file_ids)
+             |> Repo.update_all(set: [state: :active])
+         end)
+
+    # 4. Delete the inactive OwnedFiles
+    #    (Another option: Temporarily move them to another table and delete later.)
```

Comment:
Until we're confident that this is correct, I am very hesitant to actually delete the files and records.

Reply:
Good point. We may move them to another table in this PR, and when we will feel confident about this operation we may make another PR for actually deleting them.

Reply:
Done